compiler/rustc_metadata/src/dependency_format.rs

```diff
@@ -407,28 +389,19 @@ fn verify_ok(tcx: TyCtxt<'_>, list: &[Linkage]) {
             }
 
             if let Some(found_strategy) = tcx.required_panic_strategy(cnum) && desired_strategy != found_strategy {
-                sess.err(&format!(
-                    "the crate `{}` requires \
-                               panic strategy `{}` which is \
-                               incompatible with this crate's \
-                               strategy of `{}`",
-                    tcx.crate_name(cnum),
-                    found_strategy.desc(),
-                    desired_strategy.desc()
-                ));
+                sess.emit_err(RequiredPanicStrategy {
+                    crate_name: tcx.crate_name(cnum).to_string(),
+                    found_strategy: found_strategy.desc().to_string(),
```

Comment:
Can `IntoDiagnosticArg` be implemented for `PanicStrategy` that just calls `desc` on it? Then we can use the `PanicStrategy` type in the struct itself.

Reply:
✅